### PR TITLE
[SCB-118] add prometheus integration impl

### PR DESCRIPTION
**Pull request by zhengyangyong**
Signed-off-by: zhengyangyong <yangyong.zheng@huawei.com>

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
new module metrics-prometheus for integration
**two new dependency import**
```xml
      <dependency>
        <groupId>io.prometheus</groupId>
        <artifactId>simpleclient</artifactId>
        <version>0.1.0</version>
      </dependency>
      <dependency>
        <groupId>io.prometheus</groupId>
        <artifactId>simpleclient_httpserver</artifactId>
        <version>0.1.0</version>
      </dependency>
```

### CODE REVIEW

**Comment by zhengyangyong** [comment]
Rebase on upstream master

File: metrics/metrics-common/src/main/java/io/servicecomb/metrics/common/RegistryMetric.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public RegistryMetric(SystemMetric systemMetric,
 
     ConsumerInvocationMetric instanceConsumerInvocationMetric = new ConsumerInvocationMetric("instance",
         MetricsConst.INSTANCE_CONSUMER_PREFIX,
-        new TimerMetric(MetricsConst.INSTANCE_PRODUCER_PREFIX + ".producerLatency"),
+        new TimerMetric(MetricsConst.INSTANCE_CONSUMER_PREFIX + ".producerLatency"),
```
Comment (wujimin) [inline metrics/metrics-common/src/main/java/io/servicecomb/metrics/common/RegistryMetric.java]: producerLatency?

Comment (zhengyangyong) [reply metrics/metrics-common/src/main/java/io/servicecomb/metrics/common/RegistryMetric.java]: oh,no... fixed

**Comment by WillemJiang** [comment]
Do we have integration test for it? 
If not please add a JIRA for it. We need to make sure the prometheus client is working out of box.

**Comment by zhengyangyong** [comment]
Rebase on upstream master and resolve conflicts

**Comment by zhengyangyong** [comment]
Rebase on upstream master and resolve conflicts

**Pull request merged**